### ISSUE TITLE
censurfridns.dk server errors

### ISSUE THREAD

**Issue by aabdnn** [getdnsapi/stubby#16]
When running stubby with debug logging, I'm seeing:

```
[09:58:36.855948] STUBBY: 184.105.193.78                           : Conn opened: TLS - Strict Profile
[09:58:37.135059] STUBBY: 184.105.193.78                           : Conn closed   : Transport=TLS - *Failure*
[09:58:37.135227] STUBBY: 184.105.193.78                           : Conn closed: TLS - Resps=     0, Timeouts  =     0, Curr_auth =   None, Keepalive(ms)=     0
[09:58:37.135238] STUBBY: 184.105.193.78                           : Upstream   : TLS - Resps=     0, Timeouts  =     0, Best_auth =   None
[09:58:37.135247] STUBBY: 184.105.193.78                           : Upstream   : TLS - Conns=     0, Conn_fails=     1, Conn_shuts=      0, Backoffs     =     0
[09:58:37.135313] STUBBY: 184.105.193.78                           : Conn closed: TLS - Resps=     0, Timeouts  =     0, Curr_auth =   None, Keepalive(ms)=     0
[09:58:37.135327] STUBBY: 184.105.193.78                           : Upstream   : TLS - Resps=     0, Timeouts  =     0, Best_auth =   None
[09:58:37.135333] STUBBY: 184.105.193.78                           : Upstream   : TLS - Conns=     0, Conn_fails=     1, Conn_shuts=      0, Backoffs     =
```

I see the same with the IPv6 address of the server.

**Comment by saradickinson** [comment]
Hi Anand, I just noticed this too. *Completely* my fault - the addresses are wrong (they are the ones for the OARC servers!). I've just created https://github.com/getdnsapi/stubby/pull/17 which fixes this - although this uncovered an issue with parsing IPv6 addresses that end in :: which I will look into separately.

**Comment by aabdnn** [comment]
Thanks Sara. IPv6 addresses ending in a colon trip up the YAML parser, because it thinks it's the end of a key definition. You can certainly have addresses ending in a colon in YAML - just quote the address, like this:

```
- address_data: '2a01:3a0:53:53::'
```

or:

```
- address_data: "2a01:3a0:53:53::"
```

I personally prefer single quotes (saves having to press shift), but it's just a personal preference. Any type of quote will work, because YAML treats both single and double quotes the same way.

**Comment by saradickinson** [comment]
Hi,
Yaml does handle quotes, but our (limited) parser doesn't :-(  So I've taken @wtoorop suggestion of using ::0 instead in the config file. I'll convert #18 to a bug to support quotes.

**Comment by aabdnn** [comment]
@saradickinson that doesn't make sense to me. At least on OSX, stubby is linked against libyaml, and if the code calls libyaml to parse the config file, then it should handle the quotes just fine:

```
otool -L /usr/local/bin/stubby 
/usr/local/bin/stubby:
	/usr/local/opt/libyaml/lib/libyaml-0.2.dylib (compatibility version 3.0.0, current version 3.5.0)
	/usr/local/opt/getdns/lib/libgetdns.6.dylib (compatibility version 9.0.0, current version 9.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

Where, and why, are you using your own YAML parser?

**Comment by saradickinson** [comment]
Sorry - to be clearer we parse from yaml into a JSON dict which is then used to configure getdns. The yaml to json translation is what is not processing the quotes on IP addresses correctly ... 
see  src/yaml/convert_yaml_to_json.c

**Comment by saradickinson** [comment]
Closing as original issue fixed and issue #18 opened to address parsing issue. Please re-open if there is a further issue.
